Inline Podcast Bookmarks tab

bjtitus opened the pull request:
Fixes [PCIOS-126](https://linear.app/a8c/issue/PCIOS-126/bookmarks-shouldnt-open-as-modal-in-a-playlist-detail)

Moves the Bookmarks view inside the Podcast page inside of the tab.

* The `BookmarksListView` is embedded inside of a cell within the `PodcastViewController` table view. I felt that this was a compromise between keeping the functionality and style of the other bookmarks screens and avoiding a larger refactor of the `PodcastViewController`. It could use it at some point but I don't know that this is the time.
* The multi-select action bar logic was moved into `ActionBarOverlayView` and `ActionBarView` so it could be both used in the `BookmarksListView` and presented directly (and properly sized and positioned) on `PodcastViewController`. Since `BookmarksListView` is being used as a cell and can flow offscreen, the positioning isn't easily possible strictly in SwiftUI.

| | | | |
|--|--|--|--|
| <img width="1290" height="2796" alt="Simulator Screenshot - iPhone 16 Plus - 2025-09-09 at 20 58 54" src="https://github.com/user-attachments/assets/757cfb34-a892-4e0d-b321-38de3916dfdb" /> | <img width="1290" height="2796" alt="Simulator Screenshot - iPhone 16 Plus - 2025-09-09 at 20 59 08" src="https://github.com/user-attachments/assets/4b7a971f-9c1f-4ecb-8a55-ecec92aafa38" /> | <img width="1290" height="2796" alt="Simulator Screenshot - iPhone 16 Plus - 2025-09-09 at 21 00 06" src="https://github.com/user-attachments/assets/4f8aa215-a69a-42e2-887d-36e5dc41868f" /> | <img width="1290" height="2796" alt="Simulator Screenshot - iPhone 16 Plus - 2025-09-09 at 22 25 36" src="https://github.com/user-attachments/assets/aa4d68d1-75dd-4b06-b8e3-2bc7f80974f8" /> |

### Notes

I took a look at updating some of the styling to better match the Episodes tab but I think that can be a follow up to this PR to keep the size on the smaller side. I focused on maintaining complete functionality and trying to make things operate smoothly rather than making any changes to the existing design.

## To test

* Enable the `podcastBookmarksInline` Feature Flag - I've left it disabled for now as I'm not sure when this will ship and I'd like to have designers take a look first as we may want to unify a few details before launch
* Use a plus account or enable Paid Plus in the Developer menu
* Navigate to a Podcast page
* Tap the "Bookmarks" tab
* ✅ Ensure the Empty view shows up and works as expected
* Add bookmarks to a podcast, some with titles
* Check the bookmarks
* ✅ Ensure the bookmarks appear
* Search your bookmarks
* ✅ Ensure they filter as expected
* Try selecting podcasts by long tapping and using the various selection options
* Try editing and deleting bookmarks

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

bjtitus committed:
Move bookmark tab inline on Podcast page

bjtitus committed:
Fix scrolling + selection bugs with hosted list view

bjtitus committed:
Add action bar

bjtitus committed:
Fix action bar animation

bjtitus committed:
Skip editing for bookmarks tab

bjtitus committed:
Switch to contentConfiguration

bjtitus committed:
Add search field to bookmarks

bjtitus committed:
Disable multi-select when switching tabs

bjtitus committed:
Code cleanup

bjtitus committed:
Fix spacing on bookmarks

bjtitus committed:
Fix empty state content width

bjtitus committed:
Refactor: centralize mini‑player baseline inset via MiniPlayerInsets and adopt in overlay + podcast VC

bjtitus committed:
Cleanup: remove unused BookmarksList cell identifier and dead properties; unify header height logic

bjtitus committed:
Refactor: unify tab selection .onReceive to a single attachment

bjtitus committed:
Feature flag: add podcastBookmarksInline to control inline BookmarksListView in PodcastViewController

bjtitus committed:
Gate Bookmarks inline behavior in PodcastViewController.showBookmarks() behind FeatureFlag.podcastBookmarksInline

bjtitus committed:
Remove unnecessary indent handling

bjtitus committed:
Use the MiniPlayerPaddingModifier for bookmarks

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 500 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

bjtitus committed:
Refactor BookmarksHostingCell

bjtitus committed:
Swiftlint Fix

danielebogo reviewed (approved):
:shipit:

SergioEstevao commented:
I was testing the bookmarks multi-select actions and it looks they are not being triggered.

https://github.com/user-attachments/assets/908e48a5-175a-4d44-b2bf-58d40c7faf9a

SergioEstevao reviewed (changes requested):
See my comments about multi-select actions

pocketcasts commented:
<!-- REUSE_ID: 9b1cdf25-eeae-4bd7-b80a-b050ee83dd6a -->

Version `7.98` has now entered code-freeze, so the milestone of this PR has been updated to `7.99`.

bjtitus committed:
Fix bookmarks action bar bottom constraint

Before, we were offsetting the SwiftUI but this forced the view out of the bounds of the containing host view controller which caused interactions to fail.

bjtitus commented:
@SergioEstevao This is fixed now. Using the `miniPlayerSafeAreaInset` was forcing the SwiftUI view outside of the bounds of the host view controller causing the interactions to fail.

bjtitus committed:
Merge remote-tracking branch 'origin/trunk' into bjtitus/podcast-bookmark-tab-list-view

# Conflicts:
#	Modules/Utils/Sources/PocketCastsUtils/Feature Flags/FeatureFlag.swift
#	podcasts/PodcastViewController.swift

SergioEstevao commented:
@bjtitus bookmarks actions are working the only thing I notice now is this small animation "jump" when selecting the first bookmark on multi-select mode.

https://github.com/user-attachments/assets/86fb334d-b663-43f3-bcf4-825e19396b81

bjtitus committed:
Remove spacer when using external action bar

bjtitus committed:
Merge branch 'trunk' into bjtitus/podcast-bookmark-tab-list-view

# Conflicts:
#	Modules/Utils/Sources/PocketCastsUtils/Feature Flags/FeatureFlag.swift

bjtitus commented:
@SergioEstevao The jump should be fixed up:

https://github.com/user-attachments/assets/ede2341e-8c3e-4b87-915e-2c7d7ab1aa57

Can you do a sanity check when you get a chance?

SergioEstevao reviewed (approved):
Working correctly!
Thanks for all the fixes!
:shipit:
